src/Engine/Utils.java

```diff
@@ -558,4 +569,25 @@ public static boolean pathCollides(GameMap map, Unit unit, Path path)
     }
     return result;
   }
+
+  /**
+   * Evaluates the proposed move, creates a MoveEvent describing it, and adds that event to eventQueu
+   * If the move passes over an obstacle, the resulting MoveEvent will have its path shortened accordingly.
+   * @param gameMap The world in which the action is to take place.
+   * @param unit The unit who is to move.
+   * @param movePath The complete sequence of steps the unit proposes to take.
+   * @param eventQueue Will be given the new MoveEvent.
+   * @return true if the move is created as specified, false if the path was shortened.
+   */
+  public static boolean enqueueMoveEvent(GameMap gameMap, Unit unit, Path movePath, GameEventQueue eventQueue)
```

Comment:
Once again, if we're arbitrating fog interactions here, we need to force a MapMaster so we know what's what.